### PR TITLE
Add QgsPathResolver::setPathWriter and QgsPathResolver::removePathWriter

### PR DESCRIPTION
**Pull request by manisandro**
As counterparts to `QgsPathResolver::setPathPreprocessor` and `QgsPathResolver::removePathPreprocessor`.

### CODE REVIEW

**Review by nyalldawson — COMMENTED** [review]
Looking good, just a minor change re documentation

**Comment by manisandro** [comment]
Done, I believe the note that setting a new path preprocessor replaces any existing ones is actually wrong? I also added a commit to fix this.

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
>  I believe the note that setting a new path preprocessor replaces any existing ones is actually wrong?

Ah, you're right. The dox were out of date, thanks.

File: src/core/qgspathresolver.h [hunk 1]
```diff
@@ -171,6 +172,114 @@ class CORE_EXPORT QgsPathResolver
...
+     * The path writer function is called before any bad layer handler.
+     *
+     * If multiple writers are set, they will be called in sequence based on the order in which
+     * they were originally set.
+     *
+     * Example - replace path with a variable:
+     *
+     * \code{.py}
+     *   def my_processor(path):
+     *      return path.replace('c:/Users/ClintBarton/Documents/Projects', '@projectdir@')
```
Comment (DelazJ) [inline src/core/qgspathresolver.h:215]: @manisandro this does not translate correctly in the pyQGIS documentation (https://qgis.org/pyqgis/master/core/QgsPathResolver.html#qgis.core.QgsPathResolver.setPathWriter)

Comment (manisandro) [reply src/core/qgspathresolver.h:215]: So I should use i.e. $ rather than @?

Comment (DelazJ) [reply src/core/qgspathresolver.h:215]: no idea, sorry.

**Pull request merged**